sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -177,18 +181,18 @@ private void assignTraceContext(final @NotNull SentryEvent event) {
     if (event.getThrowable() != null) {
       final Pair<ISpan, String> pair = throwableToSpan.get(event.getThrowable());
       if (pair != null) {
-        if (event.getContexts().getTrace() == null) {
+        if (event.getContexts().getTrace() == null && pair.getFirst() != null) {
           event.getContexts().setTrace(pair.getFirst().getSpanContext());
         }
-        if (event.getTransaction() == null) {
+        if (event.getTransaction() == null && pair.getSecond() != null) {
           event.setTransaction(pair.getSecond());
```

Comment:
lets cache `pair.getFirst()` and `pair.getSecond()` to a var instead of calling twice, we avoid multiple lookups

Reply:
What lookup happens with these methods? Isn't this just a method call on an object? That under the hood return a reference? If that's the case, and there's no concurrency, it could be OK to just call the getter twice.